src/services/comparator.ts

```diff
@@ -0,0 +1,165 @@
+import DestinationService from '../interfaces/DestinationService';
+import { ProcessorTransformationRequest, RouterTransformationRequest, Destination } from '../types';
+import ServiceSelector from '../helpers/serviceSelector';
+import tags from '../v0/util/tags';
+import stats from '../util/stats';
+import logger from '../logger';
+import { CommonUtils } from '../util/common';
+import { PlatformError } from '../v0/util/errorTypes';
+const NS_PER_SEC = 1e9;
+
+export default class ComparatorService {
+  private static getTestThreshold(destination: Destination) {
+    return destination.DestinationDefinition?.Config['cdkV2TestThreshold'] || 0;
+  }
+
+
+  public static async compareDestinationServiceAtProc(
```

Comment:
compareDestinationServiceAtProc name appears that it is processor but inside there is logic for everything